storage/replica_command.go

```diff
@@ -1255,45 +1255,71 @@ func (r *Replica) splitTrigger(batch engine.Engine, split *proto.SplitTrigger) e
 func (r *Replica) AdminMerge(args proto.AdminMergeRequest, desc *proto.RangeDescriptor) (proto.AdminMergeResponse, error) {
 	var reply proto.AdminMergeResponse
 
-	// Lookup subsumed range.
 	if desc.EndKey.Equal(proto.KeyMax) {
-		// Noop.
-		return reply, nil
-	}
-	subsumedRng := r.rm.LookupReplica(desc.EndKey, nil)
-	if subsumedRng == nil {
-		return reply, util.Errorf("ranges not collocated; migration of ranges in anticipation of merge not yet implemented")
+		// Merging the final range doesn't make sense.
+		return reply, util.Errorf("cannot merge final range")
```

Comment:
this one too, is an unfriendly message. the grandcaller only supplied a key; perhaps we could spell out that that key is on the final range?